tree/tree/src/TTree.cxx

```diff
@@ -1209,6 +1209,18 @@ void TTree::AddClone(TTree* clone)
    }
 }
 
+void CheckIfReshuffled(TTree *t1, TTree *t2)
+{
+   const auto isT1Reshuffled = t1->TestBit(TTree::kEntriesReshuffled);
+   const auto isT2Reshuffled = t2->TestBit(TTree::kEntriesReshuffled);
+   if (isT1Reshuffled || isT2Reshuffled) {
+      const std::string reshuffledTreeName = isT1Reshuffled ? t1->GetName() : t2->GetName();
+      const auto msg = "Tree %s has kEntriesReshuffled set, and cannot be used as friend nor can be added as a friend. "
+                       "Unset the bit manually if you know what you are doing.";
+      Error("AddFriend", msg, reshuffledTreeName);
```

Comment:
const auto msg = "Tree %s has kEntriesReshuffled set, and cannot be used as friend nor can be added as a friend unless you build a Tree index on the tree %s."
      Error("AddFriend", msg, reshuffledTreeName, t2->GetName());
```

Reply:
changed, thanks!